### ISSUE TITLE
Open & copy multi-line git permalinks from multi-line code snippets

### ISSUE THREAD

**Issue by SemanticBeeng** [ben-gibson/GitLink#77]
Would be very useful to have the current functionality work for multiple line code selections and not just for one line. 

Example 
https://github.com/ben-gibson/GitLink/blob/86d191b2f7367137ecab5678f78b0081e6c1b6f3/src/uk/co/ben_gibson/git/link/Url/Factory/GitBlitUrlFactory.java#L27-L32

**Comment by ben-gibson** [comment]
I will add this in the next release.

**Comment by SemanticBeeng** [comment]
Thanks Ben. Kind share an idea of when to know if to hold my breath. :-)

Is it at all possible to have a function that given https://github.com/ben-gibson/GitLink/blob/86d191b2f7367137ecab5678f78b0081e6c1b6f3/src/uk/co/ben_gibson/git/link/Url/Factory/GitBlitUrlFactory.java#L27-L32 would return

```
        for (Map.Entry<String, String> urlPattern : patterns.entrySet()) {
            String pattern = urlPattern.getKey();
            String url = urlPattern.getValue();


            Matcher matcher = Pattern.compile(pattern).matcher(remoteUrl);
            if (matcher.matches()) {
```
I need to use it from outside the plugin much like this https://marketplace.atlassian.com/apps/1211649/gister-github-gist-macro-for-confluence?hosting=cloud&tab=overview but to link to code sections from actual projects, not just gists.

This is useful when creating books or blogs and need to embed code from actual projects and not have to copy them to gists.

Asking in this context but maybe a different request needed - please advise, thanks

**Comment by ben-gibson** [comment]
This has been released in version `3.1.0`

**Comment by SemanticBeeng** [comment]
Multi-line "Open" works fine - thanks.
But "Copy" menu does not show up for neither single or multi-line selection.
There used to be a "Copy" menu for single line, right? 

Am I missing something?
